### ISSUE TITLE
assert.Eventually() does not fail test when the timer expires

### ISSUE THREAD

**Issue by kellul** [stretchr/testify#837]
assert.Eventually() returns false (and not Fail(...)) when the timer expires.

As a result, the test does not fail.

So for example, including  `assert.Eventually(t, func()bool{return false}, time.Second, time.Millisecond))` in your test does not cause it to fail.  It will poll the function for a second, and then pass.

Is this intentional?  It seems surprising, since the other assert functions cause the test to fail when the condition is not met.

**Comment by georgelesica-wf** [comment]
If I'm not mistaken, it's the `require` version of the functions that fail, the assertions just return a boolean and print an error. For example, see: https://github.com/stretchr/testify/blob/master/require/require.go#L17

**Comment by dmacvicar** [comment]
As I was seeing the same, I did some experimentation:

```golang
package main

import (
	"testing"
	"time"
	"github.com/stretchr/testify/assert"
	//	"github.com/stretchr/testify/require"
)

func TestEventually(t *testing.T) {

	assert.Eventually(t, func() bool {
		return false
	}, 10 * time.Second, 1 * time.Second, "always false")

}
```

With v1.4.0:

```console
duncan@linux-1cur:/space/git/testify-testcase> go test .
--- FAIL: TestEventually (10.00s)
    main_test.go:12:
                Error Trace:    main_test.go:12
                Error:          Condition never satisfied
                Test:           TestEventually
                Messages:       always false
FAIL
FAIL    github.com/dmacvicar/testify-testcase   10.004s
```

With v1.4.1-0.20191106224347-f1bd0923b832

```console
duncan@linux-1cur:/space/git/testify-testcase> go test .
go: finding github.com/stretchr/testify master
ok      github.com/dmacvicar/testify-testcase   10.005s
```

That can be traced to https://github.com/stretchr/testify/commit/f1bd0923b8320ddfb0a8a9d919b5aeb8b7a4cc40 which replaced a

```golang
return Fail(t, "Condition never satisfied", msgAndArgs...)
```

with

```golang
return false
```

**Comment by kellul** [comment]
> If I'm not mistaken, it's the `require` version of the functions that fail, the assertions just return a boolean and print an error. For example, see: https://github.com/stretchr/testify/blob/master/require/require.go#L17

I don't think that's accurate.  The other `assert` functions all return `Fail(...)` when the condition is not met, which causes a test failure before returning `false`.  This one simply returns `false` with no call to `Fail(...)`.

For example, in `assert.Equal`: https://github.com/stretchr/testify/blob/master/assert/assertions.go#L339

I'm currently working around this by calling `assert.True(t, assert.Eventually(t, ...))`

It looks like @dmacvicar has already opened a PR to fix this though -- #838.  Thanks for that!
